### PR TITLE
Refactor DTO and other 3rd party framework interop tools

### PR DESCRIPTION
**Pull request by peterschutt**
# PR Checklist

- [ ] Have you followed the guidelines in `CONTRIBUTING.rst`?
- [ ] Have you got 100% test coverage on new code?
- [ ] Have you updated the prose documentation?
- [ ] Have you updated the reference documentation?

### CODE REVIEW

**Review by JacobCoffee — APPROVED** [review]

**Comment by JacobCoffee** [comment]
i like the layout

File: docs/examples/tests/application_hooks/test_application_after_exception_hook.py [hunk 1]
```diff
@@ -2,7 +2,6 @@
 from typing import TYPE_CHECKING
 
 from examples.application_hooks import after_exception_hook
-
 from starlite.testing import TestClient
```
Comment (peterschutt) [inline docs/examples/tests/application_hooks/test_application_after_exception_hook.py:5]:
```suggestion

from starlite.testing import TestClient
```

File: docs/examples/tests/application_hooks/test_application_before_send.py [hunk 2]
```diff
@@ -1,5 +1,4 @@
 from examples.application_hooks import before_send_hook
-
 from starlite.status_codes import HTTP_200_OK
```
Comment (peterschutt) [inline docs/examples/tests/application_hooks/test_application_before_send.py:2]:
```suggestion

from starlite.status_codes import HTTP_200_OK
```

File: docs/examples/tests/application_hooks/test_application_shutdown_hooks.py [hunk 3]
```diff
@@ -2,7 +2,6 @@
 from typing import TYPE_CHECKING
 
 from examples.application_hooks import shutdown_hooks
-
 from starlite.testing import TestClient
```
Comment (peterschutt) [inline docs/examples/tests/application_hooks/test_application_shutdown_hooks.py:5]:
```suggestion

from starlite.testing import TestClient
```

File: docs/examples/tests/application_hooks/test_application_startup_hooks.py [hunk 4]
```diff
@@ -2,7 +2,6 @@
 from typing import TYPE_CHECKING
 
 from examples.application_hooks import startup_hooks
-
 from starlite.testing import TestClient
```
Comment (peterschutt) [inline docs/examples/tests/application_hooks/test_application_startup_hooks.py:5]:
```suggestion

from starlite.testing import TestClient
```

File: docs/examples/tests/datastructures/headers/test_cache_control.py [hunk 5]
```diff
@@ -1,5 +1,4 @@
 from examples.datastructures.headers import cache_control
-
 from starlite.testing import TestClient
```
Comment (peterschutt) [inline docs/examples/tests/datastructures/headers/test_cache_control.py:2]:
```suggestion

from starlite.testing import TestClient
```

File: docs/examples/tests/dependency_injection/test_dependency_default_value_no_dependency_fn.py [hunk 6]
```diff
@@ -1,5 +1,4 @@
 from examples.dependency_injection import dependency_default_value_no_dependency_fn
-
 from starlite.status_codes import HTTP_200_OK
```
Comment (peterschutt) [inline docs/examples/tests/dependency_injection/test_dependency_default_value_no_dependency_fn.py:2]:
```suggestion

from starlite.status_codes import HTTP_200_OK
```

File: docs/examples/tests/dependency_injection/test_dependency_default_value_with_dependency_fn.py [hunk 7]
```diff
@@ -1,5 +1,4 @@
 from examples.dependency_injection import dependency_default_value_with_dependency_fn
-
 from starlite.status_codes import HTTP_200_OK
```
Comment (peterschutt) [inline docs/examples/tests/dependency_injection/test_dependency_default_value_with_dependency_fn.py:2]:
```suggestion

from starlite.status_codes import HTTP_200_OK
```

File: docs/examples/tests/dependency_injection/test_dependency_skip_validation.py [hunk 8]
```diff
@@ -1,5 +1,4 @@
 from examples.dependency_injection import dependency_skip_validation
-
 from starlite.status_codes import HTTP_200_OK
```
Comment (peterschutt) [inline docs/examples/tests/dependency_injection/test_dependency_skip_validation.py:2]:
```suggestion

from starlite.status_codes import HTTP_200_OK
```

File: docs/examples/tests/dependency_injection/test_dependency_validation_error.py [hunk 9]
```diff
@@ -1,5 +1,4 @@
 from examples.dependency_injection import dependency_validation_error
-
 from starlite.status_codes import HTTP_500_INTERNAL_SERVER_ERROR
```
Comment (peterschutt) [inline docs/examples/tests/dependency_injection/test_dependency_validation_error.py:2]:
```suggestion

from starlite.status_codes import HTTP_500_INTERNAL_SERVER_ERROR
```

File: docs/examples/tests/dependency_injection/tests_dependency_non_optional_not_provided.py [hunk 10]
```diff
@@ -1,6 +1,6 @@
 import pytest
-from examples.dependency_injection import dependency_non_optional_not_provided
 
+from examples.dependency_injection import dependency_non_optional_not_provided
 from starlite import Starlite
```
Comment (peterschutt) [inline docs/examples/tests/dependency_injection/tests_dependency_non_optional_not_provided.py:4]:
```suggestion
from examples.dependency_injection import dependency_non_optional_not_provided

from starlite import Starlite
```

File: docs/examples/tests/routing/test_mounting.py [hunk 11]
```diff
@@ -1,8 +1,8 @@
 from typing import TYPE_CHECKING
 
 import pytest
-from examples.routing import mount_custom_app, mounting_starlette_app
 
+from examples.routing import mount_custom_app, mounting_starlette_app
 from starlite.status_codes import HTTP_200_OK
```
Comment (peterschutt) [inline docs/examples/tests/routing/test_mounting.py:6]:
```suggestion
from examples.routing import mount_custom_app, mounting_starlette_app

from starlite.status_codes import HTTP_200_OK
```

File: docs/examples/tests/test_exceptions.py [hunk 12]
```diff
@@ -3,7 +3,6 @@
     override_default_handler,
     per_exception_handlers,
 )
```
Comment (peterschutt) [inline docs/examples/tests/test_exceptions.py:5]:
```suggestion
)

```

File: docs/examples/tests/test_hello_world.py [hunk 13]
```diff
@@ -1,5 +1,4 @@
 from examples import hello_world
-
 from starlite.status_codes import HTTP_200_OK
```
Comment (peterschutt) [inline docs/examples/tests/test_hello_world.py:2]:
```suggestion

from starlite.status_codes import HTTP_200_OK
```

File: docs/examples/tests/test_startup_and_shutdown.py [hunk 14]
```diff
@@ -2,7 +2,6 @@
 from unittest.mock import AsyncMock, MagicMock
 
 from examples import startup_and_shutdown
-
 from starlite import get
```
Comment (peterschutt) [inline docs/examples/tests/test_startup_and_shutdown.py:5]:
```suggestion

from starlite import get
```

**Pull request closed without merging**